### PR TITLE
shared src/main/resources

### PR DESCRIPTION
**Pull request by sdelamo**
This is a small change which enables us to delete a lot of duplicated files. 

Instead of having

```
guides/xxx/java
guides/xxx/java/src/main/resources
guides/xxx/kotlin
guides/xxx/kotlin/src/main/resources
guides/xxx/groovy
guides/xxx/groovy/src/main/resources
```

You will have 

```
guides/xxx/src/main/resources
guides/xxx/java
guides/xxx/kotlin
guides/xxx/groovy
```

### CODE REVIEW

**Comment by sdelamo** [comment]
if we like this change, I will apply this improvement to every guide.

File: buildSrc/src/main/groovy/io/micronaut/guides/GuideProjectGenerator.groovy [hunk 1]
```diff
@@ -145,6 +145,11 @@ class GuideProjectGenerator implements Closeable {
                 File destination = destinationPath.toFile()
                 destination.mkdir()
                 guidesGenerator.generateAppIntoDirectory(destination, app.applicationType, packageAndName, appFeatures, buildTool, testFramework, lang, javaVersion)
+
+                Path srcPath = Paths.get(inputDir.absolutePath, appName, "src")
+                if (srcPath.toFile().exists()) {
+                    Files.walkFileTree(srcPath, new CopyFileVisitor(destinationPath))
```
Comment (burtbeckwith) [inline buildSrc/src/main/groovy/io/micronaut/guides/GuideProjectGenerator.groovy]:
```suggestion
                    Files.walkFileTree(srcPath, new CopyFileVisitor(Paths.get(destination.path, 'src')))
```

**Comment by burtbeckwith** [comment]
This is great. Could you do the same for src/main/resources and src/test/resources in chess-listener?

**Comment by sdelamo** [comment]
@burtbeckwith 
> This is great. Could you do the same for src/main/resources and src/test/resources in chess-listener?

https://github.com/micronaut-projects/micronaut-guides/pull/404/commits/ab912ed4c74eb363a3d12e3be5667b6784cbe2ff

**Pull request merged**